Cattac/Cattac/GameEngine.swift

```diff
@@ -26,14 +26,17 @@ class GameEngine {
 
     // The AI engine that is used when multiplayer mode is not active
     private var gameAI: GameAI!
+    
+    /// States to advance, initialized at 1 to rollover PreStart state.
+    private var statesToAdvance: Int = 1
```

Comment:
Yeap it will be, and i agree `Bool` is more appropriate. But it will only advance state to `ActionsExecution` and stay there forever. Since the call to advance `ActionsExecution` state is merged with the call to advance `StartActionsExecution` through `true && true = true`